docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -123,32 +132,12 @@ Account contract implementation extending xref:ISRC6[`ISRC6`].
 * xref:#Account-OwnerRemoved[`++OwnerRemoved(removed_owner_guid)++`]
 --
 
-[#Account-Constructor]
-==== Constructor
-
-[.contract-item]
-[[Account-constructor]]
-==== `[.contract-item-name]#++constructor++#++(ref self: ContractState, _public_key: felt252)++` [.item-kind]#constructor#
-
-Initializes the account with the given public key, and registers the ISRC6 interface ID.
-
-Emits an {OwnerAdded} event.
-
-[#Account-External-Functions]
-==== External Functions
-
-[.contract-item]
-[[Account-__validate_deploy__]]
-==== `[.contract-item-name]#++__validate_deploy__++#++(self: @ContractState, class_hash: felt252, contract_address_salt: felt252, _public_key: felt252) → felt252++` [.item-kind]#external#
-
-Validates a https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/#deploy_account_transaction[`DeployAccount` transaction].
-See xref:/guides/deployment.adoc[Counterfactual Deployments].
-
-Returns the short string `'VALID'` if valid, otherwise it reverts.
+[#Account-Embeddable-Functions]
+==== Embeddable Functions
 
 [.contract-item]
 [[Account-__execute__]]
-==== `[.contract-item-name]#++__execute__++#++(ref self: ContractState, calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
+==== `[.contract-item-name]#++__execute__++#++(self: @ContractState, calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
```

Comment:
```suggestion
==== `[.contract-item-name]#++__execute__++#++(self: @ComponentState<TContractState>, calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
```
isn't like this now?

Reply:
Not really, the implementation that you get is not the one having ComponentState, but the one having ContractState (for embeddable ones). You don't embed the implementation we write, but the one the compiler generates from that.